doc/stake-pool-operations/KES_period.md

```diff
@@ -4,43 +4,185 @@ To create an operational certificate for a block-producing node, you need a _KES
+
+### Find the counter value used on-chain
+
+To find out the actual `OpCert` counter used on-chain, run:
+
+```bash
+> cardano-cli query kes-period-info --testnet-magic 42 --op-cert-file node-spo3/opcert.cert
+
+>
+✓ Operational certificate's KES period is within the correct KES period interval
+✓ The operational certificate counter agrees with the node protocol state counter
+{
+    "qKesCurrentKesPeriod": 15,
+    "qKesEndKesInterval": 18,
+    "qKesKesKeyExpiry": null,
+    "qKesMaxKESEvolutions": 6,
+    "qKesNodeStateOperationalCertificateNumber": 3,
+    "qKesOnDiskOperationalCertificateNumber": 3,
+    "qKesRemainingSlotsInKesPeriod": 690,
+    "qKesSlotsPerKesPeriod": 300,
+    "qKesStartKesInterval": 12
+}
+```
+
+* `"qKesNodeStateOperationalCertificateNumber": 3,` is the last counter registered **OnChain** to forge a block.
+
+* `"qKesOnDiskOperationalCertificateNumber": 3,` is the counter value of your **OnDisk** operational certificate.
+
+
+When your pool has already forged a block in the current KES interval the above values must match.
+
+If your pool has **not** yet forged a block in the current KES Interval it's ok that `cardano-cli query kes-period-info` shows that `"qKesOnDiskOperationalCertificateNumber"` is greater than `"qKesNodeStateOperationalCertificateNumber"` by exactly one (1), for example:
+
+```
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 3,
+```
+
+Once your pool forges a block `qKesNodeStateOperationalCertificateNumber` will be updated accordingly **(OnChain)** and these values will be the same.
+
+### How do I know if I have issued an _invalid_ operational certificate
+
+An easy way to know if you have issued an invalid certificate is using  `cardano-cli query kes-period-info` right after you have created the new operational certificate. If the **OnDisk** counter differs from the **OnChain** counter by more than 1, your certificate is _invalid_, for example:
+
+```
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 4,
+```
+
+### My pool has not forged a block in the current KES interval and I need to issue new operational certificate. What should I do with the counter?
+
+Keep in mind that the rule is that the counter must be one more than the last counter used to _actually_ forge a block in the _current chain_. So if we have not forged a block during the current KES interval, the counter must not be incremented when issuing a new OpCert i.e it remains the same.
+
+In this scenario `cardano-cli query kes-period-info` will show that the **OnDisk** counter is greater than the **On-Chain** counter
```

Comment:
Why in this scenario will the `cardano-cli query kes-period-info` show the ondisk counter is greater than the onchain counter?

Reply:
Did you mean to say "If, in this scenario, cardano-cli query kes-period-info show the ondisk counter is greater than the onchain counter, you cannot use `cardano-cli node issue-op-cert` because this will result in an additional increment of the op certs counter.